typing/typecore.ml

```diff
@@ -1894,12 +1894,12 @@ and type_pat_aux
   | Ppat_open (lid,p) ->
       let path, new_env =
         !type_open Asttypes.Fresh !!penv sp.ppat_loc lid in
-      penv.env <- new_env;
+      Pattern_env.set_env penv new_env;
       let p = type_pat tps category ~penv p expected_ty in
       let new_env = !!penv in
       begin match Env.remove_last_open path new_env with
       | None -> assert false
-      | Some closed_env -> penv.env <- closed_env
+      | Some closed_env -> Pattern_env.set_env penv closed_env
```

Comment:
We probably need to also revert Pattern_env.env to a referece instead of a mutable field.
Some sharing seems to be lost here compared to the previous code.

Reply:
Where ?

Reply:
[penv.env] is assigned a new value here, but the updated [penv] seems to be unused hereafter and discarded. In the previous code (before 12331), the assignment was to a reference and possibly shared.

Reply:
Okay, my bad. The sharing is fine because there is no copying of penv in the function.